scripts/bootloader/zb_add_ota_header.py

```diff
@@ -0,0 +1,204 @@
+                            file_version,
+                            zigbee_stack_version,
+                            bytes([len(header_string)]),
+                            bytes(header_string.encode('ascii')),
+                            self.__total_size]
+        self.__pack_args.extend(self.__additional_fields)
+
+    @property
+    def header(self):
+        return struct.pack(self.__pack_format, *self.__pack_args)
+
+    def __add_optional_fields(self, fields_length, field_control_bit_mask, fields_formatting, fields_values):
+        self.__total_size    += fields_length
+        self.__header_length += fields_length
+        self.__field_control += field_control_bit_mask
+        self.__pack_format   += fields_formatting
+        if type(fields_values) in (tuple, list):
+            self.__additional_fields.extend(fields_values)
+        else:
+            self.__additional_fields.append(fields_values)
+
+def convert_version_string_to_int(s):
+    """Convert from semver string "1.2.3", to integer 10203"""
+    numbers = s.split(".")
+    if len(numbers) != 3:
+        raise print("application-version-string parameter must be on the format x.y.z")
+    js = [10000, 100, 1]
+    return sum([js[i] * int(numbers[i]) for i in range(3)])
+
+def auto_int(x):
+    return int(x, 0)
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Append Zigbee OTA header to BIN file",
+        formatter_class=argparse.RawDescriptionHelpFormatter)
+
+    parser.add_argument("--application", required=True,
+                        help="The application firmware file.")
+    parser.add_argument("--application-version-string", required=True,
+                        help="The application version string, e.g. '2.7.31'. Will be converted to an integer, e.g. 207031.")
+    parser.add_argument("--zigbee-manufacturer-id", required=False, type=auto_int,
+                        help="Manufacturer ID to be used in Zigbee OTA header.")
+    parser.add_argument("--zigbee-image-type", required=False, type=auto_int,
+                        help="Image type to be used in Zigbee OTA header.")
+    parser.add_argument("--zigbee-comment", required=False, nargs='?',
+                        help="Firmware comment to be used in Zigbee OTA header.")
+    parser.add_argument("--zigbee-ota-min-hw-version", required=False, type=auto_int, nargs='?',
+                        help="The zigbee OTA minimum hw version of Zigbee OTA Client.")
+    parser.add_argument("--zigbee-ota-max-hw-version", required=False, type=auto_int, nargs='?',
+                        help="The zigbee OTA maximum hw version of Zigbee OTA Client.")
+    return parser.parse_args()
+
+
+def main():
+
+    args = parse_args()
+
+    app_file = args.application
+    app_version_int = convert_version_string_to_int(args.application_version_string)
+    zb_manufacturer_id = args.zigbee_manufacturer_id if args.zigbee_manufacturer_id is not None else OTA_HEADER_MANUFACTURER_WILDCARD
```

Comment:
As far as I remember - there is a way to provide default values to optional arguments, so you do not need this assignment.